src/nvim/api/command.c

```diff
@@ -134,11 +135,20 @@ Dict(cmd) nvim_parse_cmd(String str, Dict(empty) *opts, Arena *arena, Error *err
   char *cmdline = arena_memdupz(arena, str.data, str.size);
   const char *errormsg = NULL;
 
-  if (!parse_cmdline(cmdline, &ea, &cmdinfo, &errormsg)) {
+  int save_called_emsg = called_emsg;
+  int save_emsg_silent = emsg_silent;
+  emsg_silent++;
+  bool parse_success = parse_cmdline(cmdline, &ea, &cmdinfo, &errormsg);
+
+  emsg_silent = save_emsg_silent;
+  if (!parse_success) {
     if (errormsg != NULL) {
       api_set_error(err, kErrorTypeException, "Parsing command-line: %s", errormsg);
     } else {
-      api_set_error(err, kErrorTypeException, "Parsing command-line");
+      errormsg = get_vim_var_str(VV_ERRMSG);
+      api_set_error(err, kErrorTypeException, "%s",
+                    *errormsg != NUL
+                    && called_emsg > save_called_emsg ? errormsg : "Parsing command-line");
```

Comment:
There should still be a `Parsing command-line:` prefix, like the `errormsg != NULL` branch.